### DISCUSSION TITLE
Create an AI without an existing dataset

### DISCUSSION THREAD

**Discussion by Japhei** [Kotlin/kotlindl#351] [Q&A] [answered]
I am new to KotlinDL and neural networks and I’m trying to setup a “self-learning” AI in Kotlin.

Somehow, I can’t find a way to create my own dataset. The project needs to start with an empty dataset and evaluate the training step by step to fill it. So, there is no need of a preexisting dataset such as mnist and so on.

Specifically, I have a 12 numeric one-dimensional array (which has nothing to do with image recognition or anything like that) as input and a single float as output that represents the match of the input.

This task sounds kind of easy, but as I said, I’m new to neural networks and couldn’t find a way to get rid of an existing dataset or a way to create my own. Are there any examples provided I haven’t found?

**Comment by Japhei** [comment, marked as answer]
I just found out, that OnHeapDataset has a create function:

```
val dataset = OnHeapDataset.create(
    ::extractX,
    ::extractY
)
```

**Comment by zaleslaw** [comment]
Hi @Japhei is you question solved by yourself? Feel free to share your code example and ask any questions

**Reply by Mihail-Rain-Heart** [reply]
I would like to add to this question. You are using `extractX` and `extractY`. Why were they chosen?

If I want to implement a neural network that will transform an image (draw on it), then what kind of output from the neural network do I need? Are there any examples of this?
